### PR TITLE
Aavruti_Assignment_4

### PR DESCRIPTION
**Pull request by AavrutiDJun20**
Missing Link

**After making suggested changes:**
image

**Old Screenshot:**
image

### CODE REVIEW

File: src/aavruti/MissingLink.java [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+		WebDriver driver = new ChromeDriver();
+		driver.manage().window().maximize();
+		return driver;
+	}
+
+	void findMissingLink() {
+		WebDriver driver = driverSetup();
+		driver.get("file:///G:/TechnoGitProject/SeleniumTechnoJun20/resources/forms/MissingLink.html");
+
+		if(driver.findElement(By.xpath("//p")).getText().equals("Missing Link")) {
```
Comment (KrishnaKTechnocredits) [inline src/aavruti/MissingLink.java:27]: Here there is a no relation between p and a tag, so you should not right //a link logic under p tag.

File: src/aavruti/MissingLink.java [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+				try {
+					if(linkName.getAttribute("href").equals(""))
+						missingLink += linkName.getText();
+				}
+				catch(NullPointerException ne) {
+					hrefAttributeMissing += linkName.getText();
+				}
+			}
+			System.out.println("\nMissing Links --> " + missingLink);
+			System.out.println("href Attribute Missing links --> " + hrefAttributeMissing);
```
Comment (KrishnaKTechnocredits) [inline src/aavruti/MissingLink.java:45]: You should write this printing statement under try & catch block.

File: src/aavruti/MissingLink.java [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+	void findMissingLink() {
+		WebDriver driver = driverSetup();
+		driver.get("file:///G:/TechnoGitProject/SeleniumTechnoJun20/resources/forms/MissingLink.html");
+
+		if(driver.findElement(By.xpath("//p")).getText().equals("Missing Link")) {
+			List<WebElement> linkList = driver.findElements(By.xpath("//a"));
+			System.out.println("\n" + linkList.size() + " links on the page.");
+
+			String missingLink = "";
+			String hrefAttributeMissing = "";
```
Comment (KrishnaKTechnocredits) [inline src/aavruti/MissingLink.java:32]: This variables are not required.

File: src/aavruti/MissingLink.java (hunk 1 above)
Comment (AavrutiDJun20) [reply src/aavruti/MissingLink.java]: Removed p tag validation as per your given suggestion.

File: src/aavruti/MissingLink.java (hunk 2 above)
Comment (AavrutiDJun20) [reply src/aavruti/MissingLink.java]: Added printing statement inside try and catch block.

File: src/aavruti/MissingLink.java (hunk 3 above)
Comment (AavrutiDJun20) [reply src/aavruti/MissingLink.java]: Removed extra variables.
Thank You for suggestions.

**Review by KrishnaKTechnocredits — APPROVED** [review]
Good job.

**Pull request merged**